libafl/src/schedulers/testcase_score.rs

```diff
@@ -14,17 +14,21 @@ use crate::{
 };
 
 /// Compute the favor factor of a [`Testcase`]. Higher is better.
-pub trait TestcaseScore<I, S> {
+pub trait TestcaseScore<S>
+where
+    S: HasCorpus,
+{
     /// Computes the favor factor of a [`Testcase`]. Higher is better.
-    fn compute(state: &S, entry: &mut Testcase<I>) -> Result<f64, Error>;
+    fn compute(state: &S, entry: &mut Testcase<<S::Corpus as Corpus>::Input>)
+        -> Result<f64, Error>;
 }
```

Comment:
I'm not sure I like this. That said, `TestcaseScore` is only used with schedulers, so this is probably fine.